Formula/mtools.rb

```diff
@@ -26,6 +26,10 @@ def install
       --without-x
     ]
 
+    # The mtools configure script incorrectly detects stat64. This forces it off
+    # to fix build errors on Apple Silicon. See stat(6) and pv.rb.
+    ENV["ac_cv_func_stat64"] = "no"
```

Comment:
```suggestion
    ENV["ac_cv_func_stat64"] = "no" if Hardware::CPU.arm?
```
Should we just restrict this to ARM? Though, I guess it probably doesn't make a difference...

Reply:
It won't make a difference since we're past `stat` being transitioned from 32-bits to 64-bits. It is kind of weird that the header files for Apple Silicon are different from the x86 ones for this, though. Maybe they're slightly ahead and x86 will see the same error messages in a few months?

In any case, I'm cool with the conditional. It certainly doesn't hurt anything.

Reply:
Let's make this conditional. I've been burned by changes that were supposed to be no-op before.